Code/Framework/AzFramework/AzFramework/Device/DeviceAttributesSystemComponent.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Device/DeviceAttributesSystemComponent.h>
+#include <AzFramework/Device/DeviceAttributes.h>
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+
+namespace AzFramework
+{
+    DeviceAttributesSystemComponent::DeviceAttributesSystemComponent() = default;
+    DeviceAttributesSystemComponent::~DeviceAttributesSystemComponent() = default;
+
+    void DeviceAttributesSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<DeviceAttributesSystemComponent, AZ::Component>();
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<DeviceAttributesSystemComponent>(
+                    "AzFramework Device Attributes Component", "System component responsible for handling device attribute registration")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "Editor")
+                    ;
+            }
+        }
+    }
+
+    void DeviceAttributesSystemComponent::Activate()
+    {
+        if (DeviceAttributeRegistrar::Get() == nullptr)
+        {
+            DeviceAttributeRegistrar::Register(this);
+        }
+
+        // register default attributes
+        RegisterDeviceAttribute(AZStd::make_unique<DeviceAttributeDeviceModel>());
+        RegisterDeviceAttribute(AZStd::make_unique<DeviceAttributeRAM>());
+    }
+
+    void DeviceAttributesSystemComponent::Deactivate()
+    {
+        if (DeviceAttributeRegistrar::Get() == this)
+        {
+            DeviceAttributeRegistrar::Unregister(this);
+        }
+
+        m_deviceAttributes.clear();
+    }
+
+    bool DeviceAttributesSystemComponent::RegisterDeviceAttribute(AZStd::unique_ptr<DeviceAttribute> deviceAttribute)
+    {
+        auto name = deviceAttribute->GetName();
+        if (m_deviceAttributes.contains(name))
+        {
+            AZ_Warning(
+                "DeviceAttributesSystemComponent", false,
+                "Device attribute '%.*s' is already registered, ignoring new registration request.",
+                AZ_STRING_ARG(name));
+            return false;
+        }
+
+        m_deviceAttributes.emplace(name, AZStd::move(deviceAttribute) );
+        return true;
+    }

```

Comment:
So this logic is a bit tricky here.

Because if the device attribute is already registered, the unique_ptr containing the new instance will get deleted and the user can do nothing about it.

I recommend in this case doing two additional actions.
1. Adding a result structure that returns the unique_ptr back to the user in that case.
2. Not logging a warning here, but instead adding an error message string to the result structure and the let the caller decide to log.

```suggestion
    struct DeviceAttributeResult
    {
        explicit operator bool() const
        {
            return m_deviceAttribute != nullptr;
        }
        
        AZStd::unique_ptr<DeviceAttribute> m_deviceAttribute;
        AZStd::string m_errorMessages;
    };
    
    DeviceAttributeResult DeviceAttributesSystemComponent::RegisterDeviceAttribute(AZStd::unique_ptr<DeviceAttribute> deviceAttribute)
    {
        DeviceAttributeResult result;
        auto name = deviceAttribute->GetName();
        if (m_deviceAttributes.contains(name))
        {
            result.m_errorMessages = AZStd::string::format("Device attribute '%.*s' is already registered, ignoring new registration request.",
                AZ_STRING_ARG(name));
            result.m_deviceAttribute = AZStd::move(deviceAttribute);
            return result;
        }

        m_deviceAttributes.emplace(name, AZStd::move(deviceAttribute) );
        return result;
    }

```

Reply:
For simplicity I may just change from a unique_ptr to shared_ptr.